semver/_comparator_intersects.ts

```diff
@@ -1,54 +1,61 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 import type { Comparator } from "./types.ts";
-import { greaterOrEqual } from "./greater_or_equal.ts";
-import { lessOrEqual } from "./less_or_equal.ts";
-import { comparatorMin } from "./_comparator_min.ts";
-import { comparatorMax } from "./_comparator_max.ts";
+import { compare } from "./compare.ts";
+import { testRange } from "./test_range.ts";
+
+function comparatorIsWildcard(c: Comparator): boolean {
+  return (
+    Number.isNaN(c.major) && Number.isNaN(c.minor) && Number.isNaN(c.patch) &&
+    (c.prerelease === undefined || c.prerelease.length === 0) &&
+    (c.build === undefined || c.build.length === 0)
+  );
+}
+
 /**
- * Returns true if the range of possible versions intersects with the other comparators set of possible versions
- * @param c0 The left side comparator
- * @param c1 The right side comparator
- * @returns True if any part of the comparators intersect
+ * Returns true if the range of possible versions intersects with the other c1arators set of possible versions
+ * @param c0 The left side c1arator
+ * @param c1 The right side c1arator
+ * @returns True if any part of the c1arators intersect
```

Comment:
```suggestion
 * Returns true if the range of possible versions intersects with the other comparators set of possible versions
 * @param c0 The left side comparators
 * @param c1 The right side comparators
 * @returns True if any part of the comparators intersect
```
I'm not sure why `comparators` was changed to `c1arators`. Is this deliberate?